### ISSUE TITLE
Fail to reproduce the experiments in Appendix H

### ISSUE THREAD

**Issue by baowanxiaozi666** [Graph-and-Geometric-Learning/HyBRiD#10]
I can not reproduce  the "Model Fit Performance" in the paper.The mse indicators are too different, I added the mse indicators on the source code you provided.

**Comment by baowanxiaozi666** [comment]
This calls into question the credibility of the correlation coefficient of the CPM calculations in your paper, which is not valid for the specific task（Model-Fit）.Hope you can give the correct code

**Comment by Boltzmachine** [comment]
Could you reproduce the coefficients experiments?

Also I cannot solve your problems if you do not provide more details (e.g., how do you modify my code, and what the specific results / curves are)

**Comment by baowanxiaozi666** [comment]
> Could you reproduce the coefficients experiments?
> 
> Also I cannot solve your problems if you do not provide more details (e.g., how do you modify my code, and what the specific results / curves are)

yes, i can reproduce the coefficients experiments,but at the
same time ，i can not reproduce your mse indicators，The mean squared error (MSE) reported in your study is below 10; however, the MSE I obtained during replication is approximately 200

**Comment by baowanxiaozi666** [comment]
> > Could you reproduce the coefficients experiments?
> > Also I cannot solve your problems if you do not provide more details (e.g., how do you modify my code, and what the specific results / curves are)
> 
> yes, i can reproduce the coefficients experiments,but at the same time ，i can not reproduce your mse indicators

**Comment by baowanxiaozi666** [comment]
> > Could you reproduce the coefficients experiments?
> > Also I cannot solve your problems if you do not provide more details (e.g., how do you modify my code, and what the specific results / curves are)
> 
> yes, i can reproduce the coefficients experiments,but at the same time ，i can not reproduce your mse indicators，The mean squared error (MSE) reported in your study is below 10; however, the MSE I obtained during replication is approximately 200

**Comment by Boltzmachine** [comment]
Could you provide your code about how do you calculate / log mse?

**Comment by baowanxiaozi666** [comment]
> Could you provide your code about how do you calculate / log mse?

I use the nn.MSELoss() in pytorch to calculate the mse indicators.Because, the model you
provided has predicted the y_hat, which is
the predicted IQ

**Comment by baowanxiaozi666** [comment]
In my opinion, if the model cannot fit the downstream task, the correlation coefficient it obtains may be not reliable

**Comment by Boltzmachine** [comment]
You could use `git diff` to export the difference. That would be more convenient. Otherwise, I cannot tell what the issue is.

In regression tasks, Pearson coefficients are considered more reliable than MSE. Since the scale of MSE is uncertain and usually misleading. For example, your model can learn to cheat by always predicting the mean value with low MSE. However, Pearson coefficients avoid this situation, and are always within the range [0, 1], which makes them more reliable and interpretable.

**Comment by baowanxiaozi666** [comment]
I hope you can provide reproducible code for calculating the MSE metric.

**Comment by Boltzmachine** [comment]
I am not with my computer right now. I cannot do it right now if you want me to provide additional code. If you provide your code, I can help you solve it in a timely manner.

Besides, I log MSE in the validation step. Is the metric there normal? If it is, that could be a clue to debug.

**Comment by Boltzmachine** [comment]
Sorry I think I accidentally used
```
MSE(A, B) * (121 - 60)
```
to calculate the MSE after denormalization. But apparently it should be

```
MSE(A, B) * (121 - 60)**2
```

which means in Tables 10 & 11, **all results should be multiplied by 61 (if you want the results before normalization), or divided by 61 (if you want the results after normalization).**

Besides, an MSE ~200 is not very bad because the scale of y is large (~100), where a difference of 10% could lead to an MSE of 100.

Feel free to correct this and go with your own results you obtained. I will also correct this in the revised paper.

Here is how you could possibly calculate the MSE

```diff
diff --git a/src/model.py b/src/model.py
index 0c868ba..e6b2b46 100644
--- a/src/model.py
+++ b/src/model.py
@@ -131,6 +131,7 @@ class RegressionModule(LightningModule):
     def on_test_start(self):
         self.weights = defaultdict(list)
         self.labels = defaultdict(list)
+        self.preds = defaultdict(list)

     def get_meta_from_batch(self, batch):
         meta = batch["meta"]
@@ -152,9 +153,12 @@ class RegressionModule(LightningModule):
         dataset_name: str = meta["dataset_name"]
         self.weights[dataset_name + "-" + task].append(weights.cpu().numpy())
         self.labels[dataset_name + "-" + task].append(batch["y"].cpu().numpy())
+        self.preds[dataset_name + "-" + task].append(y_hat.cpu().numpy())

     def on_test_end(self) -> None:
+        from sklearn.metrics import mean_squared_error
         for key in self.weights.keys():
+            preds = np.concatenate(self.preds[key])
             weights = self.weights[key]
             labels = self.labels[key]
             weights = np.concatenate(weights)
@@ -172,4 +176,6 @@ class RegressionModule(LightningModule):
                 "train_id": np.asarray(train_ids),
                 "test_id": np.asarray(test_ids),
             }
+            mse = mean_squared_error(preds[test_ids], labels[test_ids])
+            print(f"MSE: after normalization {mse}, before normalization {mse * 61**2}")
             cpm(X, Y, train_test_split)
```

**Comment by baowanxiaozi666** [comment]
> Sorry I think I accidentally used对不起，我想我不小心用了
> 
> ```
> MSE(A, B) * (121 - 60)
> ```
> 
> to calculate the MSE after denormalization. But apparently it should be来计算非规范化后的 MSE。但显然应该是
> 
> ```
> MSE(A, B) * (121 - 60)**2
> ```
> 
> which means in Tables 10 & 11, **all results should be multiplied by 61 (if you want the results before normalization), or divided by 61 (if you want the results after normalization).**这意味着在表 10 和 11 中，** 所有结果都应该乘以 61（如果你想在标准化之前看到结果），或者除以 61（如果你想在标准化后的结果）。**
> 
> Besides, an MSE ~200 is not very bad because the scale of y is large (~100), where a difference of 10% could lead to an MSE of 100.此外，MSE ~200 并不是很糟糕，因为 y 的尺度很大 （~100），其中 10% 的差异可能导致 MSE 为 100。
> 
> Feel free to correct this and go with your own results you obtained. I will also correct this in the revised paper.请随时纠正这个问题，并使用您自己获得的结果。我也会在修订后的论文中纠正这一点。
> 
> Here is how you could possibly calculate the MSE以下是计算 MSE 的方法
> 
> diff --git a/src/model.py b/src/model.py
> index 0c868ba..e6b2b46 100644
> --- a/src/model.py
> +++ b/src/model.py
> @@ -131,6 +131,7 @@ class RegressionModule(LightningModule):
>      def on_test_start(self):
>          self.weights = defaultdict(list)
>          self.labels = defaultdict(list)
> +        self.preds = defaultdict(list)
> 
>      def get_meta_from_batch(self, batch):
>          meta = batch["meta"]
> @@ -152,9 +153,12 @@ class RegressionModule(LightningModule):
>          dataset_name: str = meta["dataset_name"]
>          self.weights[dataset_name + "-" + task].append(weights.cpu().numpy())
>          self.labels[dataset_name + "-" + task].append(batch["y"].cpu().numpy())
> +        self.preds[dataset_name + "-" + task].append(y_hat.cpu().numpy())
> 
>      def on_test_end(self) -> None:
> +        from sklearn.metrics import mean_squared_error
>          for key in self.weights.keys():
> +            preds = np.concatenate(self.preds[key])
>              weights = self.weights[key]
>              labels = self.labels[key]
>              weights = np.concatenate(weights)
> @@ -172,4 +176,6 @@ class RegressionModule(LightningModule):
>                  "train_id": np.asarray(train_ids),
>                  "test_id": np.asarray(test_ids),
>              }
> +            mse = mean_squared_error(preds[test_ids], labels[test_ids])
> +            print(f"MSE: after normalization {mse}, before normalization {mse * 61**2}")
>              cpm(X, Y, train_test_split)

According to the code you modified, but the normalized MSE indicator is only a few tenths of a bit, but the mse before normalized which your code provided is very close to my result before normalization

**Comment by Boltzmachine** [comment]
what does that mean? What is your result using my code?

**Comment by baowanxiaozi666** [comment]
> > Sorry I think I accidentally used对不起，我想我不小心用了
> > ```
> > MSE(A, B) * (121 - 60)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to calculate the MSE after denormalization. But apparently it should be来计算非规范化后的 MSE。但显然应该是来计算非规范化后的 MSE。但显然它应该是来计算非规范化后的 MSE。但显然应该是
> > ```
> > MSE(A, B) * (121 - 60)**2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which means in Tables 10 & 11, **all results should be multiplied by 61 (if you want the results before normalization), or divided by 61 (if you want the results after normalization).这意味着在表 10 和 11 中， 所有结果都应该乘以 61（如果你想在标准化之前看到结果），或者除以 61（如果你想在标准化后的结果）。**这意味着在表 10 和 11 中，** 所有结果都应该乘以 61（如果你想在标准化之前看到结果），或者除以 61（如果你想在标准化后的结果）。**
> > Besides, an MSE ~200 is not very bad because the scale of y is large (~100), where a difference of 10% could lead to an MSE of 100.此外，MSE ~200 并不是很糟糕，因为 y 的尺度很大 （~100），其中 10% 的差异可能导致 MSE 为 100。此外，MSE ~200 并不是很糟糕，因为 y 的尺度很大 （~100），其中 10% 的差异可能导致 MSE 为 100。此外，MSE ~200 并不是很糟糕，因为 y 的尺度很大 （~100），其中 10% 的差异可能导致 MSE 为 100。
> > Feel free to correct this and go with your own results you obtained. I will also correct this in the revised paper.请随时纠正这个问题，并使用您自己获得的结果。我也会在修订后的论文中纠正这一点。请随时纠正这个问题，并使用您自己获得的结果。我也将在修订后的论文中纠正这个问题。请随时纠正这个问题，并使用您自己获得的结果。我也会在修订后的论文中纠正这一点。
> > Here is how you could possibly calculate the MSE以下是计算 MSE 的方法以下是计算 MSE 的方法
> > diff --git a/src/model.py b/src/model.py
> > index 0c868ba..e6b2b46 100644索引 0c868ba..e6b2b46 100644
> > --- a/src/model.py
> > +++ b/src/model.py
> > @@ -131,6 +131,7 @@ class RegressionModule(LightningModule):@@ -131,6 +131,7 @@ 类 RegressionModule（LightningModule）：
> > def on_test_start(self):def on_test_start（个体经营）：
> > self.weights = defaultdict(list)self.weights = defaultdict（列表）
> > self.labels = defaultdict(list)self.labels = defaultdict（列表）
> > 
> > * ```
> >      self.preds = defaultdict(list)
> >   ```
> >   
> >   
> >       
> >         
> >       
> >   
> >         
> >       
> >   
> >       
> >     
> >   def get_meta_from_batch(self, batch):def get_meta_from_batch（self， batch）：
> >   meta = batch["meta"]  元 = batch[“元”]
> >   @@ -152,9 +153,12 @@ class RegressionModule(LightningModule):@@ -152,9 +153,12 @@ 类 RegressionModule（LightningModule）：
> >   dataset_name: str = meta["dataset_name"]dataset_name：str = meta[“dataset_name”]
> >   self.weights[dataset_name + "-" + task].append(weights.cpu().numpy())self.weights[dataset_name + “-” + 任务].append（weights.cpu（）.numpy（））
> >   self.labels[dataset_name + "-" + task].append(batch["y"].cpu().numpy())self.labels[dataset_name + “-” + 任务].append（batch[“y”].cpu（）.numpy（））
> > * ```
> >      self.preds[dataset_name + "-" + task].append(y_hat.cpu().numpy())
> >   ```
> >   
> >   
> >       
> >         
> >       
> >   
> >         
> >       
> >   
> >       
> >     
> >   def on_test_end(self) -> None:def on_test_end（self） -> None：
> > * ```
> >      from sklearn.metrics import mean_squared_error
> >      for key in self.weights.keys():
> >   ```
> > * ```
> >          preds = np.concatenate(self.preds[key])
> >          weights = self.weights[key]
> >          labels = self.labels[key]
> >          weights = np.concatenate(weights)
> >   ```
> > 
> > @@ -172,4 +176,6 @@ class RegressionModule(LightningModule):@@ -172,4 +176,6 @@ 类 RegressionModule（LightningModule）：
> > "train_id": np.asarray(train_ids),“train_id”：np.asarray（train_ids）、
> > "test_id": np.asarray(test_ids),“test_id”：np.asarray（test_ids）、
> > }
> > 
> > * ```
> >          mse = mean_squared_error(preds[test_ids], labels[test_ids])
> >   ```
> > * ```
> >          print(f"MSE: after normalization {mse}, before normalization {mse * 61**2}")
> >          cpm(X, Y, train_test_split)
> >   ```
> 
> According to the code you modified, but the normalized MSE indicator is only a few tenths of a bit, but the mse before normalized which your code provided is very close to my result before normalization。
This is also inconsistent with the description in your paper, but it is standardized

**Comment by Boltzmachine** [comment]
sorry I did not get what you mean. Could you summarize what the problems are now?

**Comment by baowanxiaozi666** [comment]
> sorry I did not get what you mean. Could you summarize what the problems are now?对不起，我没有明白你的意思。您能总结一下现在的问题是什么吗？

Using your code, after normalization, the resulting MSE indicator is around 0.05, while multiplied by 61 squared is 191, multiplied by 61 is 3.31, and the value of 3.31 is consistent with the data in your paper

**Comment by Boltzmachine** [comment]
which task are you running on?

**Comment by baowanxiaozi666** [comment]
> which task are you running on?您正在运行哪个任务？

FIQ tasks in the ABIDE dataset

**Comment by Boltzmachine** [comment]
FIQ in my paper is 3.477, and the results you got is 3.32. You think they are inconsistent. Is that what you mean?

**Comment by Boltzmachine** [comment]
Close as there had been no further activity for 3 weeks.
